src/koa-node-resolve.ts

```diff
@@ -16,31 +16,15 @@ import {resolve as resolvePath} from 'path';
 import {parse as parseURL} from 'url';
 
 import esmSpecifierTransform from './koa-esm-specifier-transform';
-import {getBasePath, noLeadingSlash} from './support/path-utils';
+import {noLeadingSlash} from './support/path-utils';
 import resolveSpecifier from './support/resolve-node-specifier';
 
-export type Options = {
-  /* On-disk package root path used for NPM package resolution; defaults to
-     current folder. */
-  packageRoot?: string,
-  /* When serving the package from a base other than root.  If this option is
-     given, no specifier resolution will be attempted for documents served
-     outside the baseHref. */
-  baseHref?: string,
-};
-
-export const middleware = (options: Options = {}): Koa.Middleware => {
-  const onDiskPackageRoot = resolvePath(options.packageRoot || '.');
-  const basePath =
-      options.baseHref ? noLeadingSlash(getBasePath(options.baseHref)) : '';
+export const middleware = (packageRoot = '.'): Koa.Middleware => {
```

Comment:
Something that just occurred to me: I might be serving multiple npm installations from my server, and so it seems that the package root would need to change depending on the request. (This is something that often happens with tachometer.)

For example, if my disk is laid out like this:

```
<server root>
├── bar
│   ├── bar.js
│   ├── node_modules/
│   └── package.json
└── foo
    ├── foo.js
    ├── node_modules/
    └── package.json
```

Then I want a request for `/foo/foo.js` to resolve relative to `<server root>/foo/foo.js`, and `/bar/bar.js` relative to `<server root>/bar/bar.js`.

So, that doesn't seem compatible with setting a fixed package root up-front. And that also makes me wonder, why not *always* use the file being served as the path from which to resolve?

Reply:
Turns out this just works as-is.  I renamed the middleware parameter to just `root` instead of `packageRoot` to eliminate the confusion.